plugin-server/src/kafka/batch-consumer.ts

```diff
@@ -208,7 +211,10 @@ export const startBatchConsumer = async ({
                 messagesProcessed += messages.length
 
                 if (autoCommit) {
-                    storeOffsetsForMessages(messages, consumer)
+                    // eslint-disable-next-line @typescript-eslint/require-await
+                    await retryIfRetriable(async () => {
+                        return storeOffsetsForMessages(messages, consumer)
```

Comment:
Is error -172 marked as retriable, or should we retry this three times whatever the error is?

Reply:
That was a good reminder to actually dig into that error some more. The error code is used in a few places, but only comes from one place when trying to store offsets:

https://github.com/confluentinc/librdkafka/blob/49f05db36e5bff78e856e33da951a5f998f9d55b/src/rdkafka_offset.h#L44-L49

https://github.com/confluentinc/librdkafka/blob/49f05db36e5bff78e856e33da951a5f998f9d55b/src/rdkafka_offset.h#L99-L102

So I actually don't think we want to bother retrying at this point in the code. I think we want to fail to store and loop back around to `consume` which will pick up new partitions. So I guess we want to catch, log and then `continue` so we don't kill the `main_loop`? I think this should only be happening if we're taking longer than the session timeout in `eachBatch`? Have we considered doing smaller batches?

As for this PR, the main thing I wanted to retry was the `ingestEvent`(i.e. `eachMessage`) calls.

Reply:
I don't think I'm happy with silently skipping the error and continuing to consume, there's a risk of us releasing a bug that creates runaway consumers that create dupes.
Let's table this one for now and focus on having the pod properly killed?